### ISSUE TITLE
In high contrast mode, settings checkboxes and button icons are invisible

### ISSUE THREAD

**Issue by ethus3h** [vector-im/element-web#9611]
Compare high contrast mode with non-high-contrast mode (top is high contrast mode):
![Screenshot_2019 05 01_20 38 59](https://user-images.githubusercontent.com/10457522/57051828-7c980880-6c51-11e9-868d-292d96677643.png)

![Screenshot_2019 05 01_20 38 26](https://user-images.githubusercontent.com/10457522/57051808-625e2a80-6c51-11e9-8596-042fb6317af9.png)

### Description

When high contrast mode is enabled in the operating system or Web browser, the settings check boxes, and the other button icons disappear.

### Steps to reproduce

- Enable high contrast mode in operating system or Web browser

Describe how what happens differs from what you expected.

Check boxes and icons are invisible, while I expect check boxes and icons to remain visible.

Log: sent/not sent?  Not sent.  

I put them at the top of the issue, since it asked for them there first.
### Version information

- **Platform**: web (in-browser) or desktop? Web browser

For the web app:

- **Browser**: Chrome, Safari, Firefox? which version? Firefox 60.5.1 ESR
- **OS**: Windows, macOS, Ubuntu, Arch Linux, etc? Gentoo GNU/Linux
- **URL**: riot.im/develop / riot.im/app / somewhere else? If a private server, what version of riot-web? riot.im/app

For the desktop app:

- **OS**: Windows, macOS, Ubuntu, Arch Linux, etc?
- **Version**: 1.x.y

**Comment by GReagle** [comment]
On a related note, the "check boxes" are not check boxes, they are toggle switches, which suck.  Can the developers make them check boxes which are very clear and understandable and easy to see in high contrast mode?

**Comment by cheryllium** [comment]
AFAIK this is because High Contrast mode overrides all background-color CSS into a high contrast color (white-on-black or black-on-white). Since the toggles rely on contrast between background colors to be visible, there is no solution without also changing how it looks when not in High Contrast mode.

A "quick" fix would be to add a solid border around both the toggle switch and the toggle ball. It's not ideal but would at least make the options visible, which is the difference between being usable and not. Something like this: 

![image](https://user-images.githubusercontent.com/5027241/79033911-1cf3ee00-7b80-11ea-8202-51253401550a.png)

Which looks like this outside of High Contrast: 
![image](https://user-images.githubusercontent.com/5027241/79033901-0c437800-7b80-11ea-9d6e-59821f1339b9.png)

And the border colors could be tweaked to make it look nicer in normal mode. (I'm not a designer.) It will show up as a contrasting color in High Contrast mode regardless of what colors we choose. 

I would support changing toggles to checkboxes as a more ideal accessibility solution. I'd be happy to play around with that option but since it is a significant UI change I'd like to hear what the core devs think first.

**Comment by Palid** [comment]
https://developer.mozilla.org/en-US/docs/Web/CSS/@media/-ms-high-contrast
https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast

**Comment by andybalaam** [comment]
This appears to have been fixed in the meantime: I can't reproduce it in current Element Web and current Firefox.

Here is me trying on Ubuntu MATE 21.10.  Note that the new high-contrast theme is also a help generally, but I is not actually the reason this is fixed - I think something must have happened some time ago to make it better.

![toggle-buttons-high-contrast](https://user-images.githubusercontent.com/76812/141095700-a764f097-102b-4acd-9c69-e8eff76c6a9b.gif)

I'll close this, but if it's still a problem on Gentoo (or some other platform) please re-open it.

**Comment by andybalaam** [comment]
Note: I had to set `layout.css.prefers-contrast.enabled` to `true` in about:config to get the OS high contrast settings to affect the theme.
